File: frontend/src/types/dataset.ts
export interface Dataset {
  id: number;
  name: string;
  description: string | null;
  file_path: string;
  num_samples: number;
  num_features: number;
  num_classes: number;
  created_at: string;
  updated_at: string | null;
  is_active: boolean;
}

export interface DatasetStats {
  dataset_id: number;
  name: string;
  total_samples: number;
  num_features: number;
  num_classes: number;
  suspicious_samples: number;
  corrected_samples: number;
  mismatched_labels: number;
  noise_percentage: number;
}

export interface Sample {
  id: number;
  dataset_id: number;
  sample_index: number;
  features: string;
  original_label: number;
  current_label: number;
  is_suspicious: boolean;
  is_corrected: boolean;
  created_at: string;
  updated_at: string | null;
}

File: backend/schemas/dataset.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List


class DatasetCreate(BaseModel):
    """Schema for creating a new dataset"""
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    label_column: Optional[str] = Field( 
        default=None,
        description="Name of the label column. Use 'auto' for auto-detection, 'last' for last column, or specify column name"
    )


class DatasetResponse(BaseModel):
    """Schema for dataset response"""
    id: int
    name: str
    description: Optional[str]
    num_samples: int
    num_features: int
    num_classes: int
    created_at: datetime
    is_active: bool
    
    class Config:
        from_attributes = True


class SampleResponse(BaseModel):
    """Schema for sample response"""
    id: int
    dataset_id: int
    sample_index: int
    current_label: int
    original_label: int
    is_suspicious: bool
    is_corrected: bool
    
    class Config:
        from_attributes = True


class DatasetStats(BaseModel):
    """Dataset statistics"""
    dataset_id: int
    total_samples: int
    num_features: int
    num_classes: int
    class_distribution: Optional[dict] = None
    
    class Config:
        from_attributes = True